### ISSUE TITLE
The depth of nested mixin is limited?

### ISSUE THREAD

**Issue by gabrielbiga** [jadejs/jade#1999]
I'm having a problem with nested mixin.

```
mixin opa(id)
  #opa(xx=id)
    block

mixin opa2(id)
  #opa(xx=id)
    block
    
mixin opa3(id)
  #opa(xx=id)
    block
    
mixin opa4(id)
  #opa(xx=id)
    block
    
mixin opa5(id)
  #opa(xx=id)
    block
    
mixin opa6(id)
  #opa(xx=id)
    block    

mixin opa7(id)
  #opa(xx=id)
    block    

+opa(12)
  +opa2(13)
    +opa3(14)
      +opa4(15)
        +opa5(16)
          | xx
          +opa6(17)
            | xx
            +opa7(18)
              | zo
```

Where is "zo"?
I'm doing wrong or its a bug?

**Comment by jeromew** [comment]
just a passing remark. is it normal that you only have 1 space between opa3 and opa4 ? It seems to me that this is not valid

**Comment by gabrielbiga** [comment]
Even adding the space, the problem remains...

**Comment by jeromew** [comment]
don't ask my why you can check on http://jade-lang.com/demo/ the compiled javascript

your bug comes from the fact that you have trailing tabs/spaces after "block" in opa6 and opa7. This leads to `block && block()` not beeing inserted in the compiled javascript

**Comment by ForbesLindesay** [comment]
Minimal example of bug

```jade
mixin m(id)
  div
    block    

+m()
  | This text should appear
```

Should render:

```html
<div>This text should appear</div>
```

but actually renders:

```html
<div></div>
```

This is a bug.

**Comment by ForbesLindesay** [comment]
The fix for this is ready and will be released as part of 2.0.0
